File: insertion_sort/ruby/insertion_sort.rb
def insertion_sort(array)
  for i in 1...(array.length)
    key = array[i]
    j = i - 1
    
    # Move elements of array[0..i-1], that are greater than key,
    # to one position ahead of their current position
    while j >= 0 and array[j] > key
      array[j + 1] = array[j]
      j -= 1
    end
    array[j + 1] = key
  end
  array
end

# Example usage
arr = [12, 11, 13, 5, 6]
sorted_array = insertion_sort(arr)
puts "Sorted array: #{sorted_array}"

File: insertion_sort/php/insertion_sort.php
<?php
function insertionSort(array &$arr)
{
    $length = count($arr);
    for ($i = 1; $i < $length; $i++) {
        $key = $arr[$i];
        $j = $i - 1;

        // Move elements of arr[0..i-1], that are greater than key, to one position ahead of their current position
        while ($j >= 0 && $arr[$j] > $key) {
            $arr[$j + 1] = $arr[$j];
            $j = $j - 1;
        }
        $arr[$j + 1] = $key;
    }
}

// Example usage:
$sampleArray = [5, 3, 8, 4, 2];
insertionSort($sampleArray);
echo "Sorted Array:\n";
print_r($sampleArray);
?>
